### PR TITLE
Dagger2, Timber, retrofit 2...

### PR DESCRIPTION
**Pull request by Zenedith**
Dagger2, Timber, retrofit 2, refresh token, updated dependencies, updated new ssl cert

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -34,6 +36,12 @@ android {
     }
 
     buildTypes {
+        debug {
+            applicationIdSuffix ".debug"
+            versionNameSuffix " Debug"
+            debuggable true
```
Comment (m4lysh) [inline app/build.gradle]: isn't debug build type debuggable by default?

File: app/src/main/java/io/vehiclehistory/injection/module/ApplicationModule.java [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+                        Timber.tag("OkHttp").d(message);
+                    }
+                });
+
+                loggingInterceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
+                okHttpClientBuilder.addInterceptor(loggingInterceptor);
+            }
+
+            return okHttpClientBuilder.build();
+        } catch (Exception e) {
```
Comment (m4lysh) [inline app/src/main/java/io/vehiclehistory/injection/module/ApplicationModule.java]: you sure?

File: app/src/main/java/io/vehiclehistory/database/SearchHistoryDb.java [hunk 3]
```diff
@@ -63,18 +65,25 @@ public Search getSearchAt(int position) {
...
-        return (int) DatabaseUtils.queryNumEntries(db, TABLE_NAME, null, null);
+        int count = (int) DatabaseUtils.queryNumEntries(db, TABLE_NAME, null, null);
+        db.close();
+        return count;
     }
 
     public void clearHistory() {
         SQLiteDatabase db = historyDbHelper.getWritableDatabase();
         db.delete(TABLE_NAME, null, null);
+        db.close();
```
Comment (m4lysh) [inline app/src/main/java/io/vehiclehistory/database/SearchHistoryDb.java:86]: +100

File: app/src/main/java/io/vehiclehistory/data/api/caller/BaseCaller.java [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+    public BaseCaller(
+            DataManager dataManager,
+            Retrofit retrofit
+    ) {
+        this.dataManager = dataManager;
+        this.retrofit = retrofit;
+    }
+
+    private T mMvpView;
```
Comment (m4lysh) [inline app/src/main/java/io/vehiclehistory/data/api/caller/BaseCaller.java]: remove m, move to other fields

File: app/src/main/java/io/vehiclehistory/data/AuthProvider.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    }
+
+    @Override
+    public void onPasswordFailed() {
+        //toast TODO
+    }
+
+    @Override
+    public void onNoConnectionError() {
+
```
Comment (m4lysh) [inline app/src/main/java/io/vehiclehistory/data/AuthProvider.java]: nop or handle

File: app/src/main/java/io/vehiclehistory/components/ObscuredSharedPreferences.java [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+    protected String decrypt(String value) {
+        try {
+            final byte[] bytes = value != null ? Base64.decode(value, Base64.DEFAULT) : new byte[0];
+            SecretKeyFactory keyFactory = SecretKeyFactory.getInstance("PBEWithMD5AndDES");
+            SecretKey key = keyFactory.generateSecret(new PBEKeySpec(SECRET));
+            Cipher pbeCipher = Cipher.getInstance("PBEWithMD5AndDES");
+            pbeCipher.init(Cipher.DECRYPT_MODE, key, new PBEParameterSpec(Settings.Secure.getString(context.getContentResolver(), Settings.System.ANDROID_ID).getBytes(UTF8), 20));
+            return new String(pbeCipher.doFinal(bytes), UTF8);
+
+        } catch (Exception e) {
```
Comment (m4lysh) [inline app/src/main/java/io/vehiclehistory/components/ObscuredSharedPreferences.java]: you sure? How about multicatch?

File: app/build.gradle (hunk 1 above)
Comment (Zenedith) [inline app/build.gradle]: Sure, it is

File: app/src/main/java/io/vehiclehistory/injection/module/ApplicationModule.java (hunk 2 above)
Comment (Zenedith) [inline app/src/main/java/io/vehiclehistory/injection/module/ApplicationModule.java]: ?

File: app/src/main/java/io/vehiclehistory/data/AuthProvider.java (hunk 5 above)
Comment (Zenedith) [inline app/src/main/java/io/vehiclehistory/data/AuthProvider.java]: done

File: app/src/main/java/io/vehiclehistory/data/api/caller/BaseCaller.java (hunk 4 above)
Comment (Zenedith) [inline app/src/main/java/io/vehiclehistory/data/api/caller/BaseCaller.java]: done

File: app/src/main/java/io/vehiclehistory/components/ObscuredSharedPreferences.java (hunk 6 above)
Comment (Zenedith) [inline app/src/main/java/io/vehiclehistory/components/ObscuredSharedPreferences.java]: done

**Pull request closed without merging**